third-party/realdds/src/topics/image-msg.cpp

```diff
@@ -34,6 +49,38 @@ image::create_topic( std::shared_ptr< dds_participant > const & participant, cha
 }
 
 
+/*static*/ bool
+image::take_next( dds_topic_reader & reader, image * output, eprosima::fastdds::dds::SampleInfo * info )
+{
+    raw::device::image raw_data;
+    eprosima::fastdds::dds::SampleInfo info_;
+    if ( !info )
+        info = &info_;  // use the local copy if the user hasn't provided their own
+    auto status = reader->take_next_sample( &raw_data, info );
+    if ( status == ReturnCode_t::RETCODE_OK )
+    {
+        // We have data
+        if ( output )
+        {
+            // Only samples for which valid_data is true should be accessed
+            // valid_data indicates that the instance is still ALIVE and the `take` return an
+            // updated sample
+            if ( !info->valid_data )
+                output->invalidate();
+            else
+                *output = std::move( raw_data ); //TODO - optimize copy, use dds loans
+        }
+
+        return true;
+    }
+    if ( status == ReturnCode_t::RETCODE_NO_DATA )
+    {
+        // This is an expected return code and is not an error
+        return false;
+    }
+    DDS_API_CALL_THROW( "device_info::take_next", status );
```

Comment:
device_info

Reply:
Are the quautes necessary? Just checking, I dont remember this API calls..

Reply:
Fixed. Quotes necessary.